Allow the user to enter package tracking details. Mark III: Yosemite

ctarda opened the pull request:
This is the third of a few PRs which will add support for allowing users to enter package tracking details manually, with the help of the Shipment Tracking extension in WC.

This PR contains changes to the Yosemite layer.

Ref: #689

Testing

Checkout branch and build. Check that there are no errors in the console (i.e. coredata migration errors)

Run the tests

 If there are user facing changes, I have added an item to RELEASE-NOTES.txt.

bummytime commented on Yosemite/YosemiteTests/Stores/ShipmentStoreTests.swift:
```diff
@@ -202,6 +216,148 @@ class ShipmentStoreTests: XCTestCase {
+        let group = DispatchGroup()
+
+        group.enter()
+        shipmentStore.upsertTrackingProviderDataInBackground(siteID: sampleSiteID,
```
This write to Core Data should probably happen outside of the `DispatchGroup()` — I was seeing a race condition occur while debugging.

See my test branch → https://github.com/woocommerce/woocommerce-ios/blob/try/689-mark3-bummy/Yosemite/YosemiteTests/Stores/ShipmentStoreTests.swift#L285

ctarda commented on Networking/NetworkingTests/Remote/ShipmentsRemoteTests.swift:
```diff
@@ -370,7 +370,7 @@ final class ShipmentsRemoteTests: XCTestCase {
         let expectation = self.expectation(description: "Load shipment tracking providers information")
 
-        network.simulateResponse(requestUrlSuffix: "orders/\(sampleOrderID)/shipment-trackings/providers", filename: "shipment_tracking_providers")
+        network.simulateResponse(requestUrlSuffix: "orders/\(sampleOrderID)/shipment-trackings/providers", filename: "shipmentt")
```
🤦‍♂️

ctarda commented:
Thanks for unblocking the PR @bummytime !

I incorporated your feedback and added methods to create trackings. I think this is finally ready for a proper review now 😅

ctarda commented:
Thank *you* @bummytime . I've learnt a lot about CoreData (and about myself 😛 ) working on this PR. 🙇
